### PR TITLE
edtlib: add lookup table for nodes from their dependency ordinals

### PR DESCRIPTION
**Pull request by mbolivar-nordic**
Reworked version of https://github.com/zephyrproject-rtos/zephyr/commit/f2c0e648c0baa4679e4335514114dac26c699881.

Taken from / needed by #26616.

### CODE REVIEW

**Review by pabigot — APPROVED** [review]
Typo in the commit message of edtlib: add EDT.depord2node: s/looup/lookup/

Otherwise looks great; thanks.

**Comment by mbolivar-nordic** [comment]
> Typo in the commit message of edtlib

Fixed, thanks.

**Review by tbursztyka — APPROVED** [review]

File: scripts/dts/edtlib.py [hunk 1]
```diff
@@ -134,6 +134,10 @@ class EDT:
       A collections.OrderedDict that maps a node label to the node with
       that label.
 
+    depord2node:
```
Comment (galak) [inline scripts/dts/edtlib.py:137]: Can we come up with a better name?  Or dep_ord2node?

**Review by galak — COMMENTED** [review]
LGTM, one query if we can find a better name for depord2node.

File: scripts/dts/edtlib.py (hunk 1 above)
Comment (pabigot) [reply scripts/dts/edtlib.py:137]: I don't care what it's named.

Comment (mbolivar-nordic) [reply scripts/dts/edtlib.py]: Rebased and renamed `dep_ord2node`.

**Comment by mbolivar-nordic** [comment]
The pytest-ified test suite seems to have exposed cycles in our test case devicetrees that wasn't showing up before for some reason. I added a patch at the beginning of the series to break the cycles.

**Review by galak — APPROVED** [review]

**Pull request merged**